onnxruntime/core/framework/compute_capability.h

```diff
@@ -12,22 +12,23 @@ struct ComputeCapability {
   // or multiple nodes.
   std::unique_ptr<IndexedSubGraph> sub_graph;
 
-  // When an execution provider fuses a subgraph into a kernel, it passes
-  // a kernel create function to onnxruntime so the runtime can create the
-  // compute kernel for the subgraph. Otherwise onnxruntime will search
-  // kernels in pre-defined kernel registry provided by XP.
-  KernelCreateFn fuse_kernel_function;
+  // When an execution provider fuses a subgraph into a kernel,
+  // there are two ways to run the subgraph: with a predefined kernel,
+  // or with code-generation that compile the subgraph at runtime.
+  // if this flag is true, execution provider's Compile method will be invoked
+  // if onnxruntime decide to assign the subgraph to execution provider.
+  bool need_compile;
```

Comment:
I am kind of prefer user explicit declare use pre-defined kernel or compile, otherwise some thirdparty custom kernel may accidently break the assumption. but I am open for that.

Reply:
synced offline, let's remove this flag for now.